runtime/lua/vim/lsp.lua

```diff
@@ -805,11 +806,26 @@ function lsp.start_client(config)
     }
     local version = vim.version()
 
-    if config.root_dir and not config.workspace_folders then
-      config.workspace_folders = {{
-        uri = vim.uri_from_fname(config.root_dir);
-        name = string.format("%s", config.root_dir);
-      }};
+    local workspace_folders
+    local root_uri
+    local root_path
+    if config.workspace_folders or config.root_dir then
+      -- Legacy compatibility, will be removed in neovim 0.7
+      if config.root_dir and not config.workspace_folders then
+        vim.notify("Calling start_client with root_dir is deprecated in nvim 0.6 and will be removed in 0.7, please see `:help start_client`", vim.log.levels.WARN)
```

Comment:
```suggestion
        vim.notify("Calling start_client with root_dir is deprecated and will be removed in 0.7, use workspace_folders instead, see `:help start_client`", vim.log.levels.WARN)
```

I think giving a hint about what to do can't hurt